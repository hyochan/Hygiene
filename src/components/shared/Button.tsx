import {
  ActivityIndicator,
  TextStyle,
  TouchableOpacity,
  TouchableOpacityProps,
  ViewStyle,
} from 'react-native';

import React from 'react';
import styled from 'styled-components/native';

const StyledButton = styled.View`
  align-self: center;
  /* width: 320px; */
  height: 52px;
  border-color: blue;
  align-items: center;
  justify-content: center;
  flex-direction: row;
`;

const StyledDisabled = styled(StyledButton)`
  background-color: #cccccc;
  border-color: rgb(200,200,200);
`;

const StyledText = styled.Text`
  font-size: 16px;
  color: #069ccd;
`;

const StyledDisabledText = styled(StyledText)`
  color: #969696;
`;

interface Props {
  testID?: string;
  isLoading?: boolean;
  isDisabled?: boolean;
  onPress?: () => void;
  containerStyle?: ViewStyle;
  style?: ViewStyle;
  disabledStyle?: ViewStyle;
  textStyle?: TextStyle;
  disabledTextStyle?: TextStyle;
  leftElement?: React.ReactElement;
  rightElement?: React.ReactElement;
  indicatorColor?: string;
  activeOpacity?: number;
  text?: string;
  touchableOpacityProps?: TouchableOpacityProps;
}

function Button(props: Props): React.ReactElement {
  const {
    isDisabled,
    testID,
    disabledStyle,
    containerStyle,
    isLoading,
    style,
    textStyle,
    disabledTextStyle,
    indicatorColor,
    text,
    leftElement,
    rightElement,
    activeOpacity,
    onPress,
    touchableOpacityProps,
  } = props;

  if (isDisabled) {
    return (
      <StyledDisabled
        testID={testID}
        style={[
          containerStyle,
          disabledStyle,
        ]}
      >
        <StyledDisabledText
          style={[textStyle, disabledTextStyle]}
        >{text}</StyledDisabledText>
      </StyledDisabled>
    );
  }

  if (isLoading) {
    return (
      <StyledButton testID={testID} style={[
        containerStyle,
        style,
      ]}>
        <ActivityIndicator
          size="small"
          color={indicatorColor}
        />
      </StyledButton>
    );
  }

  return (
    <TouchableOpacity
      testID={testID}
      activeOpacity={activeOpacity}
      onPress={onPress}
      style={containerStyle}
      delayPressIn={30}
      {...touchableOpacityProps}
    >
      <StyledButton style={style}>
        {leftElement || null}
        <StyledText style={textStyle}>{text}</StyledText>
        {rightElement || null}
      </StyledButton>
    </TouchableOpacity>
  );
}

export default Button;
