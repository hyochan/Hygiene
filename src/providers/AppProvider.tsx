import React, { useReducer } from 'react';

import { User } from '../types';
import createCtx from '../utils/createCtx';

interface Context {
  state: State;
  setAppLoading: (loading: boolean) => void;
  setUser: (user: User) => void;
  setUserPoint: (point: number) => void;
  resetUser: () => void;
  callDefault: () => void;
}

const [useCtx, Provider] = createCtx<Context>();

export enum ActionType {
  SetAppLoading = 'set-app-loading',
  ResetUser = 'reset-user',
  SetUser = 'set-user',
  SetUserPoint = 'set-user-point',
  CallDefault = 'call-default',
}

export interface State {
  appLoading?: boolean;
  user: User | null;
}

const initialState: State = {
  user: null,
};

interface SetAppLoadingAction {
  type: ActionType.SetAppLoading;
  payload: boolean;
}

interface SetUserAction {
  type: ActionType.SetUser;
  payload: User;
}

interface SetUserPointAction {
  type: ActionType.SetUserPoint;
  payload: number;
}

interface ResetUserAction {
  type: ActionType.ResetUser;
}

interface GetStateAction {
  type: ActionType.CallDefault;
}

type Action = SetAppLoadingAction | SetUserAction | SetUserPointAction | ResetUserAction | GetStateAction;

interface Props {
  children?: React.ReactElement;
}

type Reducer = (state: State, action: Action) => State;

const callDefault = (dispatch: React.Dispatch<GetStateAction>) => (): void => {
  dispatch({
    type: ActionType.CallDefault,
  });
};

const setAppLoading = (dispatch: React.Dispatch<SetAppLoadingAction>) => (
  loading: boolean,
): void => {
  dispatch({
    type: ActionType.SetAppLoading,
    payload: loading,
  });
};

const setUser = (dispatch: React.Dispatch<SetUserAction>) => (
  user: User,
): void => {
  dispatch({
    type: ActionType.SetUser,
    payload: user,
  });
};

const setUserPoint = (dispatch: React.Dispatch<SetUserPointAction>) => (
  point: number,
): void => {
  dispatch({
    type: ActionType.SetUserPoint,
    payload: point,
  });
};

const resetUser = (dispatch: React.Dispatch<ResetUserAction>) => (): void => {
  dispatch({
    type: ActionType.ResetUser,
  });
};

const reducer: Reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'reset-user':
      return initialState;
    case 'set-user':
      return { ...state, user: action.payload };
    case 'set-user-point':
      if (!state.user) {
        return state;
      }

      return {
        ...state,
        user: {
          ...state.user,
          point: action.payload,
        },
      };
    case 'set-app-loading':
      return { ...state, appLoading: action.payload };
    default:
      return state;
  }
};

function AppProvider(props: Props): React.ReactElement {
  const [state, dispatch] = useReducer<Reducer>(reducer, initialState);

  const actions = {
    setAppLoading: setAppLoading(dispatch),
    setUser: setUser(dispatch),
    setUserPoint: setUserPoint(dispatch),
    resetUser: resetUser(dispatch),
    callDefault: callDefault(dispatch),
  };

  return <Provider value={{ state, ...actions }}>{props.children}</Provider>;
}

export { useCtx as useAppContext, AppProvider };
