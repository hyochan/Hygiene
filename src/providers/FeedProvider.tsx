import React, { useState } from 'react';

import { Activity } from '../types';
import createCtx from '../utils/createCtx';

interface Context {
  feeds: Activity[];
  setFeeds: React.Dispatch<React.SetStateAction<Activity[]>>;
}

const [useCtx, Provider] = createCtx<Context>();

interface Props {
  children?: React.ReactElement;
}

function FeedProvider({ children }: Props): React.ReactElement {
  const [feeds, setFeeds] = useState<Activity[]>([]);

  return (
    <Provider
      value={{
        feeds,
        setFeeds,
      }}
    >
      {children}
    </Provider>
  );
}

export { useCtx as useFeedContext, FeedProvider };
