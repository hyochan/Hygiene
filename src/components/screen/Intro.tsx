import * as AppleAuthentication from 'expo-apple-authentication';
import * as Crypto from 'expo-crypto';
import * as WebBrowser from 'expo-web-browser';

import { Alert, Linking, Platform, ScrollView, TouchableOpacity, View } from 'react-native';
import { IC_BOTTLE, SvgApple, SvgFacebook, SvgGoogle } from '../../utils/Icons';
import React, { ReactElement, useState } from 'react';
import { facebookAppId, facebookSecret, googleClientId, googleSecret } from '../../../config';

import Button from '../shared/Button';
import { SocialAuthProvider } from '../../types';
import SocialSignInButton from '../shared/SocialSignInButton';
import { createUser } from '../../services/firebaseService';
import firebase from 'firebase/app';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useAppContext } from '../../providers/AppProvider';
import { useThemeContext } from '../../providers/ThemeProvider';

const Container = styled.SafeAreaView`
  flex: 1;
  justify-content: center;
  background: ${({ theme }): string => theme.background};
`;

const Wrapper = styled.View`
  margin: 20px 40px;
`;

const LogoWrapper = styled.View`
  margin-top: 200px;
  margin-bottom: 72px;
  align-self: center;
`;

const StyledLogoImage = styled.Image`
  width: 124px;
  height: 170px;
`;

const SocialButtonWrapper = styled.View`
  margin-bottom: 24px;
`;

const StyledAgreementTextWrapper = styled.View`
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  padding: 0 0 40px 0;
`;

const StyledAgreementText = styled.Text`
  line-height: 22px;
  color: #777;
`;

const StyledAgreementLinedText = styled.Text`
  line-height: 22px;
  color: ${({ theme }): string => theme.link};
  text-decoration-line: underline;
`;

export default function Intro({ navigation }): ReactElement {
  const { theme, changeThemeType } = useThemeContext();
  const { setUser } = useAppContext();

  const [signingInApple, setSigningInApple] = useState<boolean>(false);

  WebBrowser.maybeCompleteAuthSession();

  const goToWebView = (uri: string): void => {
    navigation.navigate('WebView', { uri });
  };

  const appleLogin = async (): Promise<void> => {
    setSigningInApple(true);
    try {
      const csrf = Math.random().toString(36).substring(2, 15);
      const nonce = Math.random().toString(36).substring(2, 10);
      const hashedNonce = await Crypto.digestStringAsync(
        Crypto.CryptoDigestAlgorithm.SHA256, nonce);
      const appleCredential = await AppleAuthentication.signInAsync({
        requestedScopes: [
          AppleAuthentication.AppleAuthenticationScope.FULL_NAME,
          AppleAuthentication.AppleAuthenticationScope.EMAIL,
        ],
        state: csrf,
        nonce: hashedNonce,
      });
      const { identityToken, email, state } = appleCredential;

      if (identityToken) {
        const provider = new firebase.auth.OAuthProvider('apple.com');
        const credential = provider.credential({
          idToken: identityToken,
          rawNonce: nonce, // nonce value from above
        });
        const authResult = await firebase.auth().signInWithCredential(credential);
        const user = await createUser(authResult);
        if (user) {
          // ensure update to navigate when this is new user
          setUser(user);
        }
      }
    } catch (e) {
      if (e.code === 'ERR_CANCELED') {
        // handle that the user canceled the sign-in flow
      } else {
        if (Platform.OS === 'web') {
          // @ts-ignore
          alert(`Apple Login Error: ${e.code} - ${e.message}`);
          return;
        }
        Alert.alert(`Apple Login Error: ${e.code} - ${e.message}`);
      }
    } finally {
      setSigningInApple(false);
    }
  };

  return (
    <Container>
      <ScrollView style={{ alignSelf: 'stretch' }}>
        <Wrapper>
          <LogoWrapper>
            <TouchableOpacity
              style={{
                width: 124,
              }}
              testID="theme-test"
              onPress={(): void => changeThemeType()}
            ><StyledLogoImage source={IC_BOTTLE} /></TouchableOpacity>
          </LogoWrapper>
          <SocialButtonWrapper>
            {
              Platform.select({
                ios: <Button
                  testID="btn-apple"
                  style={{
                    backgroundColor: theme.appleBackground,
                    borderColor: theme.appleText,
                    width: '100%',
                    height: 48,
                    borderWidth: 1,
                    marginBottom: 12,
                    borderRadius: 100,
                  }}
                  leftElement={
                    <View style={{ marginRight: 6 }}>
                      <SvgApple width={18} height={18} fill={theme.appleIcon}/>
                    </View>
                  }
                  isLoading={signingInApple}
                  indicatorColor={theme.primary}
                  onPress={appleLogin}
                  text={getString('SIGN_IN_WITH_APPLE')}
                  textStyle={{ fontWeight: '700', color: theme.appleText }}
                />,
              })
            }
            <SocialSignInButton
              clientId={facebookAppId}
              clientSecret={facebookSecret}
              svgIcon={<SvgFacebook width={18} height={18} fill={theme.facebookIcon}/>}
              onUserCreated={(user): void => {
                if (user) setUser(user);
              }}
              socialProvider={SocialAuthProvider.Facebook}
            />
            <SocialSignInButton
              clientId={googleClientId}
              clientSecret={googleSecret}
              svgIcon={<SvgGoogle width={20} height={20} fill={theme.googleIcon}/>}
              onUserCreated={(user): void => {
                if (user) setUser(user);
              }}
              socialProvider={SocialAuthProvider.Google}
            />
          </SocialButtonWrapper>
          <StyledAgreementTextWrapper>
            <StyledAgreementText>{getString('AGREEMENT1')}</StyledAgreementText>
            <StyledAgreementLinedText
              testID="btn-terms"
              onPress={(): void => {
                const url = 'https://dooboolab.com/termsofservice';
                if (Platform.OS === 'web') {
                  Linking.openURL(url);
                  return;
                }
                goToWebView(url);
              }}
            >
              {getString('AGREEMENT2')}
            </StyledAgreementLinedText>
            <StyledAgreementText>{getString('AGREEMENT3')}</StyledAgreementText>
            <StyledAgreementLinedText
              testID="btn-privacy"
              onPress={(): void => {
                const url = 'https://dooboolab.com/privacyandpolicy';
                if (Platform.OS === 'web') {
                  Linking.openURL(url);
                  return;
                }
                goToWebView(url);
              }}
            >
              {getString('AGREEMENT4')}
            </StyledAgreementLinedText>
            <StyledAgreementText>{getString('AGREEMENT5')}</StyledAgreementText>
          </StyledAgreementTextWrapper>
        </Wrapper>
      </ScrollView>
    </Container>
  );
}
