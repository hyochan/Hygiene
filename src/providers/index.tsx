import { ThemeProvider, ThemeType } from '../providers/ThemeProvider';

import { ActionSheetProvider } from '@expo/react-native-action-sheet';
import { AppProvider } from './AppProvider';
import { FeedProvider } from './FeedProvider';
import React from 'react';
import { useColorScheme } from 'react-native-appearance';

interface Props {
  initialThemeType?: ThemeType;
  children?: React.ReactElement;
}

// Add providers here
const RootProvider = ({
  initialThemeType = ThemeType.LIGHT,
  children,
}: Props): React.ReactElement => {
  const colorScheme = useColorScheme();

  return (
    <ThemeProvider
      initialThemeType={
        colorScheme === 'dark'
          ? ThemeType.DARK
          : colorScheme === 'light'
            ? ThemeType.LIGHT : initialThemeType}
    >
      <ActionSheetProvider>
        <AppProvider>
          <FeedProvider>
            {children}
          </FeedProvider>
        </AppProvider>
      </ActionSheetProvider>
    </ThemeProvider>
  );
};

export default RootProvider;
