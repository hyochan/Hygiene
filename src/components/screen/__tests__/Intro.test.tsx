import React, { ReactElement } from 'react';
import { RenderAPI, act, fireEvent, render } from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Button from '../../shared/Button';
import Intro from '../Intro';
import { ThemeType } from '../../../providers/ThemeProvider';
import renderer from 'react-test-renderer';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

describe('[Intro] screen rendering test', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Intro {...props} />);
    testingLib = render(component);
  });

  it('should render outer component and snapshot matches', () => {
    const json = renderer.create(component).toJSON();

    expect(json).toMatchSnapshot();
    expect(json).toBeTruthy();
  });

  it('should render [Dark] theme', () => {
    component = createTestElement(<Intro {...props} />, ThemeType.DARK);
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});
