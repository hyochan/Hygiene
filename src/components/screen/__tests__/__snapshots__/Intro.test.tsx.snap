// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Intro] screen rendering test should render [Dark] theme 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <SafeAreaView
        style={
          Array [
            Object {
              "backgroundColor": "#191919",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <ScrollView
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginBottom": 20,
                  "marginLeft": 40,
                  "marginRight": 40,
                  "marginTop": 20,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "marginBottom": 72,
                    "marginTop": 200,
                  },
                ]
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                style={
                  Object {
                    "width": 124,
                  }
                }
                testID="theme-test"
              >
                <Image
                  source={
                    Object {
                      "canInstrument": true,
                      "createTransformer": [Function],
                      "getCacheKey": [Function],
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 170,
                        "width": 124,
                      },
                    ]
                  }
                />
              </TouchableOpacity>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginBottom": 24,
                  },
                ]
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                delayPressIn={30}
                testID="btn-apple"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "borderColor": "blue",
                        "flexDirection": "row",
                        "height": 52,
                        "justifyContent": "center",
                      },
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#151E22",
                        "borderRadius": 100,
                        "borderWidth": 1,
                        "height": 48,
                        "marginBottom": 12,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "marginRight": 6,
                      }
                    }
                  >
                    <SvgMock
                      fill="#151E22"
                      height={18}
                      width={18}
                    />
                  </View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#069ccd",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#151E22",
                          "fontWeight": "700",
                        },
                      ]
                    }
                  >
                    Sign in with Apple
                  </Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                delayPressIn={30}
                testID="btn-facebook"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "borderColor": "blue",
                        "flexDirection": "row",
                        "height": 52,
                        "justifyContent": "center",
                      },
                      Object {
                        "backgroundColor": "#345997",
                        "borderColor": "#345997",
                        "borderRadius": 100,
                        "borderWidth": 1,
                        "height": 48,
                        "marginBottom": 12,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "marginRight": 6,
                      }
                    }
                  >
                    <SvgMock
                      fill="#ffffff"
                      height={18}
                      width={18}
                    />
                  </View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#069ccd",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#ffffff",
                          "fontWeight": "700",
                        },
                      ]
                    }
                  >
                    Sign in with Facebook
                  </Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                delayPressIn={30}
                testID="btn-google"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "borderColor": "blue",
                        "flexDirection": "row",
                        "height": 52,
                        "justifyContent": "center",
                      },
                      Object {
                        "backgroundColor": "#E04238",
                        "borderColor": "#E04238",
                        "borderRadius": 100,
                        "borderWidth": 1,
                        "height": 48,
                        "marginBottom": 12,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "marginRight": 6,
                      }
                    }
                  >
                    <SvgMock
                      fill="#ffffff"
                      height={20}
                      width={20}
                    />
                  </View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#069ccd",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#ffffff",
                          "fontWeight": "700",
                        },
                      ]
                    }
                  >
                    Sign in with Google
                  </Text>
                </View>
              </TouchableOpacity>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "center",
                    "paddingBottom": 40,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#777",
                      "lineHeight": 22,
                    },
                  ]
                }
              >
                We consider that you agree with 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#e2f2ff",
                      "lineHeight": 22,
                      "textDecorationLine": "underline",
                    },
                  ]
                }
                testID="btn-terms"
              >
                Terms for Agreement
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#777",
                      "lineHeight": 22,
                    },
                  ]
                }
              >
                 and 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#e2f2ff",
                      "lineHeight": 22,
                      "textDecorationLine": "underline",
                    },
                  ]
                }
                testID="btn-privacy"
              >
                Privacy Policy
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#777",
                      "lineHeight": 22,
                    },
                  ]
                }
              >
                 by going onto next step.
              </Text>
            </View>
          </View>
        </ScrollView>
      </SafeAreaView>
    </View>
  </View>
</View>
`;

exports[`[Intro] screen rendering test should render outer component and snapshot matches 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <SafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <ScrollView
      style={
        Object {
          "alignSelf": "stretch",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginBottom": 20,
              "marginLeft": 40,
              "marginRight": 40,
              "marginTop": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "marginBottom": 72,
                "marginTop": 200,
              },
            ]
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "width": 124,
              }
            }
            testID="theme-test"
          >
            <Image
              source={
                Object {
                  "canInstrument": true,
                  "createTransformer": [Function],
                  "getCacheKey": [Function],
                  "process": [Function],
                }
              }
              style={
                Array [
                  Object {
                    "height": 170,
                    "width": 124,
                  },
                ]
              }
            />
          </TouchableOpacity>
        </View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 24,
              },
            ]
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            delayPressIn={30}
            onPress={[Function]}
            testID="btn-apple"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "blue",
                    "flexDirection": "row",
                    "height": 52,
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderColor": "#151E22",
                    "borderRadius": 100,
                    "borderWidth": 1,
                    "height": 48,
                    "marginBottom": 12,
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 6,
                  }
                }
              >
                <SvgMock
                  fill="#151E22"
                  height={18}
                  width={18}
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#069ccd",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#151E22",
                      "fontWeight": "700",
                    },
                  ]
                }
              >
                Sign in with Apple
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            delayPressIn={30}
            onPress={[Function]}
            testID="btn-facebook"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "blue",
                    "flexDirection": "row",
                    "height": 52,
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#345997",
                    "borderColor": "#345997",
                    "borderRadius": 100,
                    "borderWidth": 1,
                    "height": 48,
                    "marginBottom": 12,
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 6,
                  }
                }
              >
                <SvgMock
                  fill="#ffffff"
                  height={18}
                  width={18}
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#069ccd",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#ffffff",
                      "fontWeight": "700",
                    },
                  ]
                }
              >
                Sign in with Facebook
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            delayPressIn={30}
            onPress={[Function]}
            testID="btn-google"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "blue",
                    "flexDirection": "row",
                    "height": 52,
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#E04238",
                    "borderColor": "#E04238",
                    "borderRadius": 100,
                    "borderWidth": 1,
                    "height": 48,
                    "marginBottom": 12,
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 6,
                  }
                }
              >
                <SvgMock
                  fill="#ffffff"
                  height={20}
                  width={20}
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#069ccd",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#ffffff",
                      "fontWeight": "700",
                    },
                  ]
                }
              >
                Sign in with Google
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "center",
                "paddingBottom": 40,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#777",
                  "lineHeight": 22,
                },
              ]
            }
          >
            We consider that you agree with 
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "lineHeight": 22,
                  "textDecorationLine": "underline",
                },
              ]
            }
            testID="btn-terms"
          >
            Terms for Agreement
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#777",
                  "lineHeight": 22,
                },
              ]
            }
          >
             and 
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "lineHeight": 22,
                  "textDecorationLine": "underline",
                },
              ]
            }
            testID="btn-privacy"
          >
            Privacy Policy
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#777",
                  "lineHeight": 22,
                },
              ]
            }
          >
             by going onto next step.
          </Text>
        </View>
      </View>
    </ScrollView>
  </SafeAreaView>
</View>
`;
